src/ripple/json/impl/json_reader.cpp

```diff
@@ -924,9 +923,8 @@ Reader::getLocationLineAndColumn(Location location) const
 {
     int line, column;
     getLocationLineAndColumn(location, line, column);
-    char buffer[18 + 16 + 16 + 1];
-    sprintf(buffer, "Line %d, Column %d", line, column);
-    return buffer;
+    return "Line " + std::to_string(line) + ", Column " +
+        std::to_string(column);
```

Comment:
This does multiple allocations. Consider using snprintf instead. Something like: 
```
    int line, column;
    getLocationLineAndColumn(location, line, column);
    std::size_t n = 18 + 16 + 16 + 1;
    char buffer[n];
    snprintf(buffer, n, "Line %d, Column %d", line, column);
    return buffer;
```

Reply:
Okay. I tried to do this in my first commit. But @a-noni-mousse  had reservations with this approach (https://github.com/XRPLF/rippled/pull/4570#discussion_r1229908127)

Should I still go ahead with the snprintf approach?

Reply:
This isn't a performance critical function, the extra allocations aren't _that_ big a deal. On the other hand, I don't find the `snprintf` dangerous here - it's clearly not going to overwrite the buffer and it's very easy to see that. So the issues I'd weigh are:

* 4 memory allocations vs 1 memory allocation
* a high level implementation vs a low level implementation
* this is not a performance critical function, and is unlikely to ever be performance critical

It's your call, I'll approve this with either interface.

Reply:
Hmm okay, thanks for the comparison.

I feel the `snprintf` adds complexity. I don't understand why we need to allocate `18 + 16 + 16 + 1` as the buffer size (I mimicked the existing code's behavior)

I'm inclined the favor the `std::string` version for it's simplicity.